frontend/src/main.rs

```diff
@@ -0,0 +1,197 @@
+use syntect::highlighting::{Theme, ThemeSet};
+use syntect::html::highlighted_html_for_string;
+use syntect::parsing::{SyntaxReference, SyntaxSet};
+use wasm_bindgen::JsCast;
+use web_sys::{window, Element, HtmlTextAreaElement};
+use yew::prelude::*;
+pub mod message;
+pub mod response;
+use crate::message::Msg;
+use crate::response::{Res, SpecificResponse};
+pub struct Main {
+    code: String,
+    ss: SyntaxSet,
+    theme: Theme,
+    syntax: SyntaxReference,
+    html: String,
+    div: Element,
+    code_response: Res,
+}
+impl Component for Main {
+    type Message = Msg;
+    type Properties = ();
+
+    fn create(_ctx: &Context<Self>) -> Self {
+        let code = "\n".repeat(20);
+        let ss = SyntaxSet::load_defaults_newlines();
+        let ts = ThemeSet::load_defaults();
+        let theme = ts.themes["base16-ocean.dark"].clone();
+        let syntax = ss.find_syntax_by_extension("rs").unwrap().to_owned();
+        let html = highlighted_html_for_string(&code, &ss, &syntax, &theme).expect("Can't parse");
+        let div = web_sys::window()
+            .unwrap()
+            .document()
+            .unwrap()
+            .create_element("div")
+            .unwrap();
+        div.set_inner_html(&html);
+        Self {
+            code,
+            ss,
+            theme,
+            syntax,
+            html,
+            div,
+            code_response: Res {
+                errors: SpecificResponse {
+                    compile: String::from(""),
+                    run: String::from(""),
+                },
+                outputs: SpecificResponse {
+                    compile: String::from(""),
+                    run: String::from(""),
+                },
+            },
+        }
+    }
+
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            Msg::SetContent(content) => {
+                self.code = content + "\n";
+                self.html =
+                    highlighted_html_for_string(&self.code, &self.ss, &self.syntax, &self.theme)
+                        .expect("Can't parse");
+                self.div.set_inner_html(&self.html);
+                true
+            }
+            Msg::SendCode => {
+                let code = self.code.clone();
+                _ctx.link().send_message(Msg::SetResponse(Res {
```

Comment:
nazwy zaczynające się od _ to zwyczajowo nazwy dla zmiennych niewykorzystywanych